### PR TITLE
New Workflow for Demo App

### PR DESCRIPTION
**Pull request by ugoamanoh**
## What's here?
Task URL: https://www.pivotaltracker.com/story/show/161593974

This adds a new home screen for the demo app to allow for separating functionality and use cases of the SDK. These will include 
- Create an Account
- Manage Keys
- Create a JWT
- Verify a JWT
- Selective Disclosure
 
## Screenshots
screenshot_1541088317

## Testing
Run demo app
Other tests to be added.

## Note
This branch will only cover the implementation of the home screen and creating an account.

### CODE REVIEW

File: demoapp/src/main/java/me/uport/sdk/demoapp/MainListActivity.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+class MainListActivity : AppCompatActivity() {
+
+    val features = arrayOf("Create an Account", "Manage Keys", "Create a JWT", "Verify a JWT", "Selective Disclosure")
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_list_main)
+
+        var listView = findViewById<ListView>(R.id.feature_list)
```
Comment (mirceanis) [inline demoapp/src/main/java/me/uport/sdk/demoapp/MainListActivity.kt]: generally `val` is preferred over `var`

In this case, the demoapp is already using the `kotlin-android-extensions` plugin which makes `findViewById` obsolete.
You can simply use something like:
 ```kotlin
import kotlinx.android.synthetic.main.activity_list_main.* //this import is added automagically by Android Studio

...
feature_list.adapter = adapter
feature_list.setOnItemClickListener {...}
```

File: demoapp/src/main/java/me/uport/sdk/demoapp/MainListActivity.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        listView.adapter = adapter
+
+        listView.setOnItemClickListener { _, _, position, _ ->
+            itemSelected(position)
+        }
+    }
+
+    private fun itemSelected(position: Int) {
+        when(position) {
+            0 -> startActivity(Intent(this, CreateAccountActivity::class.java))
```
Comment (mirceanis) [inline demoapp/src/main/java/me/uport/sdk/demoapp/MainListActivity.kt]: I think it's worth adding a `default` option that shows a "not implemented yet" toast message for all the options that are missing

File: demoapp/src/main/java/me/uport/sdk/demoapp/MainListActivity.kt (hunk 1 above)
Comment (ugoamanoh) [reply demoapp/src/main/java/me/uport/sdk/demoapp/MainListActivity.kt]: Nice! no more boiler plate

File: demoapp/src/main/java/me/uport/sdk/demoapp/MainListActivity.kt (hunk 2 above)
Comment (ugoamanoh) [reply demoapp/src/main/java/me/uport/sdk/demoapp/MainListActivity.kt]: Alright

**Pull request merged**